HW06.Rmd

```diff
@@ -1,92 +1,181 @@
+  dplyr::select(gene_symbol, ends_with("_H3K27M")) %>%
+  gather(key = "file", value = "value", -gene_symbol) %>%
+  spread(key = gene_symbol, value = value)
+
+#Now, to perform the t test, create empty vectors to store p values and t values
+Genes <- colnames(WT_table)
+p_value <- c()
+t_stat <- c()
+
+#Then, with a for loop, iterate across all columns (i.e. genes), starting from column 2 (column 1 = "file"). 
+for(i in 2:ncol(WT_table))
+{
+  p_value[i] <- t.test(WT_table[,i],H3K27M_table[,i])$p.value
+  t_stat[i] <- t.test(WT_table[,i],H3K27M_table[,i])$statistic
+}
+
+#Lastly, combine the vectors to create a data table with gene_id, p value and t statistic values
+t_results <- data.frame(Genes,p_value,t_stat)[-1,]
+
+#print out the t-test results
+knitr::kable(t_results, format = "markdown", col.names = c("Gene", "p-value","t statistic"))
+```
+
+Now create a graphing function to create boxplots to visualize the results. Plot expression on the y-axis. The graph should look like [this example](IRX1_example.png)
+
+```{r create a graphing, function }
+#to work in the tidyverse, it will be easier to make tidy the dataframe first
+
+#What I did first, was to create a new column in both H3K27M and WT tables. This column will have the information on mutation status, which will be useful when both tables will be merged (to be able to later use as categorical variable for the graphing function)
+H3K27M_table$Class <-as.factor("H3K27M")
+WT_table$Class <-as.factor("WT")
+
+#Merge tables
+table <- bind_rows(H3K27M_table,WT_table)
+
+#Graphing function
+#For this function, it is only necessary to specify the gene_id (columns in data frame, see below). Then is a simple function to create a boxplot similar to the one in the example. I used aes_string instead of aes so it can identify 'gene_id' in each iteration. The categorical variable (x axis and color) is "Class" the variable that I created at the beggining of the chunk.
+
+figures_FUN <- function(gene_id = "column_name"){
+  ggplot(table, aes_string(x = "Class", y = gene_id, fill = "Class"))+
+    geom_boxplot() +
+    labs(x = "H3K27M mutation" ,y = "Expression (TPM)",title=paste(gene_id, "Expression in GBM models by H3K27 mutated or WT status")) +
+    theme_bw() +
+    theme(legend.position = "none") 
+  }
+
+#then use a for loop combined with the graphing function to make a graph for all your genes of interest 
+
+#To be able to use the function within a for loop it is necessary to have a vector with gene_ids (this corresponds to the column names of the table, excluding the "file" and "Clas" variables)
+genes <- colnames(table[,-c(1,28)])
+
+#Then, to create a new directory where plots will be stored
+dir.create("plots") 
+
+#For loop to save .png files with the figures for all genes (to the directory created above)
+for(j in 1:length(genes))
+{
+  figures_FUN(gene_id = genes[j])
+  ggsave(filename=paste0(genes[j],"_plot.png"),plot = last_plot(),device="png",path="./plots")
+}
```

Comment:
This worked like a charm! I will for sure be using this general method the next time I need to make a bunch of similar plots!